recipes/ms-gsl/all/conanfile.py

```diff
@@ -52,3 +54,5 @@ def package_id(self):
         
     def package_info(self):
         self.cpp_info.defines = [self._contract_map[str(self.options.on_contract_violation)]]
+        self.cpp_info.names["cmake_find_package"] = "Microsoft.GSL"
+        self.cpp_info.names["cmake_find_package_multi"] = "Microsoft.GSL"
```

Comment:
TODO: conan should create cmake target `Microsoft.GSL::GSL` according to upstream:
```cmake
find_package(Microsoft.GSL CONFIG)
add_executable(main main.cpp)
target_link_libraries(main PRIVATE Microsoft.GSL::GSL)
```

Reply:
https://github.com/microsoft/GSL#usage-in-cmake

Reply:
@uilianries is target name creation now possible with conan 1.24?

Reply:
Not yet. It's part of Components feature, but it's under development: https://github.com/conan-io/conan/pull/6871

Reply:
The package naming changed in version 3, would it be possible leave the old naming for the v2.0.0. ie different naming for each?

Reply:
Which name? Did they change only the name or also the project? For example, Catch and Catch2 are separated packages.

Reply:
To be clear, in v2.0.0 there was no name or export target from their CMake

https://github.com/microsoft/GSL/blob/v2.0.0/CMakeLists.txt#L75-L78 just installed the headers

Reply:
Okay, but cmake_find_package generates a cmake file. Again, Conan cmake files != upstream cmake files.

Reply:
It was to be otherwise the it should be called conan_find_package...

Having always such differences makes people think that conan does not work, has bugs and then they look for alternatives